pkg/cmd/server/start/start_master.go

```diff
@@ -586,6 +586,7 @@ func startControllers(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) erro
 		kc.RunPersistentVolumeClaimBinder(binderClient)
 		kc.RunPersistentVolumeProvisioner(provisionerClient)
 		kc.RunPersistentVolumeClaimRecycler(oc.ImageFor("recycler"), recyclerClient, oc.Options.PolicyConfig.OpenShiftInfrastructureNamespace)
+		kc.RunGCController()
```

Comment:
Modified `RunGCController` to match others by starting the controller in a goroutine. Issue should be resolved now.